src/extension/csw/core/src/main/java/org/geoserver/csw/GetCapabilities.java

```diff
@@ -278,6 +278,8 @@ public GetCapabilities(CSWInfo csw, CatalogStore store, ApplicationContext conte
     }
 
     public CapabilitiesType run(GetCapabilitiesType request) throws CSWException {
+        CSWInfo expandedCsw = (CSWInfo) csw.clone(true);

```

Comment:
Thinking out loud here, instead of spreading changes like this all over the place, making all the code aware of the parametrization, isn't it better to have GeoServer.getService return a cloned object for all situation where we know in advance we are not editing configuration? Like, a special thread local could be set to inform GeoServerImpl that the current request is a REST or a GUI one, and maybe we could have a public method to set the mode (expanded or parametric) for any other custom code that needs to work in parametric mode.

To make a parallel with security, we could have made all the code security aware, but it would have been a massive maintenance chore, instead we apply security transparently. I'm hoping parametrization could be done the same way.

I've checked the proposal, it was generally talking about xstreamloaders (which would have been a very centralized point, only a few classes) but would not have allowed to edit the params via the UI (so I understand why a different path was attempted)